save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/userprofile/UserProfileView.kt

```diff
@@ -0,0 +1,272 @@
+import com.saveourtool.save.frontend.TabMenuBar
+import com.saveourtool.save.frontend.components.tables.*
+import com.saveourtool.save.frontend.components.views.contests.tab
+import com.saveourtool.save.frontend.externals.fontawesome.*
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.frontend.utils.noopLoadingHandler
+import com.saveourtool.save.info.UserInfo
+import com.saveourtool.save.v1
+import com.saveourtool.save.validation.FrontendRoutes
+import js.core.jso
+import org.w3c.fetch.Headers
+import react.*
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.figure
+import react.dom.html.ReactHTML.h1
+import react.dom.html.ReactHTML.img
+import react.dom.html.ReactHTML.td
+import react.router.dom.Link
+import web.cssom.*
+
+val userProfileView: FC<UserProfileViewProps> = FC { props ->
+    useBackground(Style.WHITE)
+
+    val (user, setUser) = useState<UserInfo?>(null)
+    val (selectedMenu, setSelectedMenu) = useState(UserProfileTab.VULNERABILITIES)
+
+    useRequest {
+        val userNew: UserInfo = get(
+            "$apiUrl/users/${props.userName}",
+            Headers().apply {
+                set("Accept", "application/json")
+            },
+            loadingHandler = ::noopLoadingHandler,
+        )
+            .decodeFromJsonString()
+
+        setUser(userNew)
+    }
+
+    @Suppress(
+        "TYPE_ALIAS",
+        "MAGIC_NUMBER",
+    )
+    val vulnerabilityTable: FC<TableProps<VulnerabilityDto>> = tableComponent(
+        columns = {
+            columns {
+                column(id = "name", header = "Name", { name }) { cellContext ->
+                    Fragment.create {
+                        td {
+                            Link {
+                                to = "/${FrontendRoutes.FOSS_GRAPH}/${cellContext.row.original.name}"
+                                +cellContext.value
+                            }
+                        }
+                    }
+                }
+                column(id = "short_description", header = "Description", { progress }) { cellContext ->
+                    Fragment.create {
+                        td {
+                            +cellContext.row.original.shortDescription
+                        }
+                    }
+                }
+                column(id = "progress", header = "Criticality", { progress }) { cellContext ->
+                    Fragment.create {
+                        td {
+                            +"${ cellContext.row.original.progress }"
+                        }
+                    }
+                }
+            }
+        },
+        initialPageSize = 10,
+        useServerPaging = false,
+        usePageSelection = false,
+        isTransparentGrid = true,
+    )
+
+    div {
+        className = ClassName("row justify-content-center")
+
+        // ===================== LEFT COLUMN =======================================================================
```

Comment:
please split it to extension methods of ChildrenBuilder

Reply:
It might be even better to split it to `VFC`/`FC`

Reply:
Fixed